File: src/66.加一.cs
/*
 * @lc app=leetcode.cn id=66 lang=csharp
 *
 * [66] 加一
 */

// @lc code=start
using System.Collections.Generic;
public class Solution
{
    public int[] PlusOne(int[] digits)
    {
        for (int i = digits.Length - 1; i >= 0; i--)
        {
            if (digits[i] == 9)
            {
                digits[i] = 0;
            }
            else
            {
                digits[i] += 1;
                return digits;
            }
        }
        var list = new List<int>() { 1 };
        foreach (int item in digits)
        {
            list.Add(0);
        }
        return list.ToArray();
    }
}
// @lc code=end

File: src/66.加一.java
import java.util.ArrayList;
import java.util.List;

/*
 * @lc app=leetcode.cn id=66 lang=java
 *
 * [66] 加一
 */

// @lc code=start
class Solution {
    public int[] plusOne(int[] digits) {
        for (int i = digits.length - 1; i >= 0; i--)
        {
            if (digits[i] == 9)
            {
                digits[i] = 0;
            }
            else
            {
                digits[i] += 1;
                return digits;
            }
        }
        int[] res = new int[digits.length + 1];
        res[0] = 1;
        for (int j = 1; j < res.length; j++) {
            res[j] = 0;
        }
        return res;
    }
}
// @lc code=end

